src/backend/joanie/tests/core/test_api_course_run.py

```diff
@@ -0,0 +1,281 @@
+"""Test suite for the CourseRun API"""
+from joanie.core import factories, models
+from joanie.tests.base import BaseAPITestCase
+
+
+class CourseRunApiTest(BaseAPITestCase):
+    """Test the API of the CourseRun resource."""
+
+    def test_api_course_run_read_list_anonymous(self):
+        """
+        It should not be possible to retrieve the list of course runs
+        for anonymous users.
+        """
+        factories.CourseRunFactory()
+
+        response = self.client.get("/api/course-runs/")
+
+        self.assertContains(
+            response,
+            "The requested resource was not found on this server.",
+            status_code=404,
+        )
+
+    def test_api_course_run_read_list_authenticated(self):
+        """
+        It should not be possible to retrieve the list of course runs
+        for authenticated users.
+        """
+        factories.CourseRunFactory()
+        user = factories.UserFactory.build()
```

Comment:
```suggestion
        user = factories.UserFactory()
```
I find that curious to use a method here and not the line over. Usually, we don't call build() just the constructor

Reply:
Instead of `create`, `build` do not persist resource into database. `UserFactory()` is an alias of `UserFactory.create()`